### PR TITLE
TLS config in the operator

### LINKED ISSUE #9780: SSL config in the operator
* Add a new field fo the Keycloak CRD for specifying a [TLS Secret](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets) name.
* Mount the Secret and configure Keycloak to use it.
* Update the example KC deployment.

Follow-up tasks:
* #9781
* #9784

### LINKED ISSUE #9782: Use prod profile in Quarkus dist
The operator currently deploys Keycloak using dev profile. We should use prod profile by default and make sure auto-build is disabled.

Prerequisite:
* #9605

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves #9780
Resolves #9782 ( specifically: ppl can use the `start-dev` command using the `unsupported` `podTemplate` feature )

### CODE REVIEW

File: operator/src/main/resources/base-keycloak-deployment.yaml [hunk 1]
```diff
@@ -27,16 +28,26 @@ spec:
         - containerPort: 8080
           protocol: TCP
         livenessProbe:
-          httpGet:
```
Comment (andreaTP) [inline operator/src/main/resources/base-keycloak-deployment.yaml:30]: `httpGet` doesn't support the "insecure" option

File: operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java [hunk 2]
```diff
@@ -136,23 +136,25 @@ public void testKeycloakCacheIsConnected() throws Exception {
                         .pods()
                         .inNamespace(namespace)
                         .withName(pod.getMetadata().getName())
-                        .portForward(8080, 8080)) {
+                        .portForward(8443, 8443)) {
```
Comment (andreaTP) [inline operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java:139]: unfortunately we cannot get rid of the port-forwarding thing, it still doesn't work even with strict hostname.

**Review by pedroigor — DISMISSED** [review]

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java [hunk 3]
```diff
@@ -181,7 +181,7 @@ protected void savePodLogs() {
 
   private static void setDefaultAwaitilityTimings() {
     Awaitility.setDefaultPollInterval(Duration.ofSeconds(1));
-    Awaitility.setDefaultTimeout(Duration.ofSeconds(240));
+    Awaitility.setDefaultTimeout(Duration.ofSeconds(360));
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java:184]: yes, at least seems like.

File: operator/src/test/java/org/keycloak/operator/utils/K8sUtils.java [hunk 4]
```diff
@@ -69,16 +75,28 @@ public static void deployDefaultKeycloak(KubernetesClient client) {
...
+                                .inNamespace(kc.getMetadata().getNamespace())
+                                .withName(kc.getMetadata().getName())
+                                .get();
+
+                        CRAssert.assertKeycloakStatusCondition(currentKc, KeycloakStatusCondition.READY, true);
+                        CRAssert.assertKeycloakStatusCondition(currentKc, KeycloakStatusCondition.HAS_ERRORS, false);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        throw e;
+                    }
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/utils/K8sUtils.java:94]: not a leftover, I intentionally left it here for easier debugging.

> the stacktrace is printed if awaitility fails

no, it doesn't maybe we are missing a configuration or something?

File: operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (vmuzikar) [reply operator/src/test/java/org/keycloak/operator/ClusterOperatorTest.java:184]: Do you know why? Kinda seems like a red flag.

File: operator/src/test/java/org/keycloak/operator/utils/K8sUtils.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/utils/K8sUtils.java:94]: anyhow, the underlying problem is the stability, I remove it for now.

**Comment by andreaTP** [comment]
@vmuzikar ready for next round I believe.

File: operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java [hunk 5]
```diff
@@ -145,4 +148,78 @@ public void testExtensions() {
         }
     }
 
+    @Test
+    public void testTlsDisabled() {
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java:176]: here you go:
https://github.com/keycloak/keycloak/pull/10605/files#diff-6c9bb3849a00b8bf0ad94a582971f6335e9cefadd546187ac395cace155b2ecaR167

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks @andreaTP.

JFTR. The only missing thing is watching the TLS Secret for changes to restart the Deployment. I'll sneak it in as part of #9726.

We can merge.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Oh, one last thing. Please correctly refer to #9782 so it's closed with this PR. Multiple issues can be linked to a single PR.

**Comment by andreaTP** [comment]
@vmuzikar done

**Comment by andreaTP** [comment]
force pushed only to re-trigger the CI, @vmuzikar do you mind to re-approve?

**Comment by vmuzikar** [comment]
Operator tests failed. Not sure if related to this change.

**Review by pedroigor — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
LGTM, assuming the tests pass.

**Comment by vmuzikar** [comment]
All green! Can you please merge @abstractj?

**Pull request merged**